gk/main.c

```diff
@@ -1164,13 +1221,27 @@ static void
 destroy_gk_lpm(struct gk_lpm *ltbl)
 {
 	destroy_ipv4_lpm(ltbl->lpm);
+	ltbl->lpm = NULL;
+
+	if (ltbl->fib_tbl != NULL) {
+		rte_free(ltbl->fib_tbl);
+		ltbl->fib_tbl = NULL;
```

Comment:
Doesn't `rte_free()` do nothing when its parameter is `NULL`? If so, you can drop the `if` test here and in the IPv6 below.